server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -263,7 +263,7 @@
  */
 public class SearchModule {
     public static final Setting<Integer> INDICES_MAX_CLAUSE_COUNT_SETTING = Setting.intSetting("indices.query.bool.max_clause_count",
-            1024, 1, Integer.MAX_VALUE, Setting.Property.NodeScope);
+            2048, 1, Integer.MAX_VALUE, Setting.Property.NodeScope);
```

Comment:
I think this should be 4096?

Reply:
I went back and forth between 2048 and 4096, thanks for catching!